wolfssl/wolfcrypt/port/Espressif/esp32-crypt.h

```diff
@@ -67,19 +188,72 @@
     #else
         #include <rom/ets_sys.h>
     #endif
-
+    #define ESP_PROHIBIT_SMALL_X 0
+#elif defined(CONFIG_IDF_TARGET_ESP32S2)
+    #include "soc/dport_reg.h"
+    #include "soc/hwcrypto_reg.h"
+    #if defined(ESP_IDF_VERSION_MAJOR) && ESP_IDF_VERSION_MAJOR >= 5
+        #include "esp_private/periph_ctrl.h"
+    #else
+        #include "driver/periph_ctrl.h"
+    #endif
+    #define ESP_PROHIBIT_SMALL_X 0
+#elif defined(CONFIG_IDF_TARGET_ESP32S3)
+    #include "soc/dport_reg.h"
+    #include "soc/hwcrypto_reg.h"
+    #if defined(ESP_IDF_VERSION_MAJOR) && ESP_IDF_VERSION_MAJOR >= 5
+        #include "esp_private/periph_ctrl.h"
+    #else
+        #include "driver/periph_ctrl.h"
+    #endif
+    #define ESP_PROHIBIT_SMALL_X 0
+#elif defined(CONFIG_IDF_TARGET_ESP32C3)
+    /* no includes for ESP32C3 at this time (no HW implemented yet) */
+#else
+    /* not yet supported. no HW */
 #endif
 
+#if defined(USE_ESP_DPORT_ACCESS_READ_BUFFER)
+    #define ESP_NO_ERRATA_MITIGATION
+#endif
 
 #ifdef __cplusplus
-    extern "C" {
+extern "C"
+{
 #endif
 
-int esp_ShowExtendedSystemInfo(void);
+    /*
+    ******************************************************************************
+    ** Some common esp utilities
+    ******************************************************************************
+    */
+
+    int esp_ShowExtendedSystemInfo(void);
+
+    /* Compare MATH_INT_T A to MATH_INT_T B
+     * During debug, the strings name_A and name_B can help
+     * identify variable name. */
+    int esp_mp_cmp(char* name_A, MATH_INT_T* A, char* name_B, MATH_INT_T* B);
```

Comment:
we should control the visibility of these API, if used only in the wolfSSL library than please use WOLFSSL_LOCAL

Reply:
hm. Interesting. Good idea. Do you think this is something an end user would ever want to use, or is it just for our own internal diagnostics and testing?

I think I will make it `WOLFSSL_LOCAL` until someone requests it, eh?

Reply:
Better to error on the side of WOLFSSL_LOCAL. Please do this though for all functions in the .h file that are only used locally. Keeping a clear boundary of what are public API and private API is good.

Reply:
Agreed. I suppose _in theory_  someone _may_ want to call some function such as a stand-alone `esp_mp_mul()` operation, but I think that would be rare, if ever. We could address that in a future request. I've flagged everything in `esp32-crypt.h` as `WOLFSSL_LOCAL` since the intention is really for acceleration to be used only in wolfcrypt. 

This will be included when I push my pending changes.